cluster.go

```diff
@@ -149,14 +149,35 @@ type ClusterConfig struct {
 	// If not provided, a default dialer configured with ConnectTimeout will be used.
 	Dialer Dialer
 
+	// Scylla-specific cluster options
+	Scylla ScyllaClusterOpts
+
 	// internal config for testing
 	disableControlConn bool
 }
 
+// Scylla-specific cluster options
```

Comment:
I do not think that users should be concerned with it, it should work out of the box if supported by node.

Reply:
For instance the way it is now you must must change your app in addition to bumping the driver version.

Reply:
I agree that it would be better if it worked out of the box, however I see one case where it could cause problems.

Consider a scenario in which the client application updates the driver and the cluster so that they both support shard-aware port, but the shard-aware port is not reachable from the client. The driver will establish a single connection to each node, but will constantly fail to fill connection pools because it will try to make new connections to the shard-aware port which is unreachable. Host pool filling is asynchronous and errors are not directly reported to the user (unless they use something like a ConnectObserver), so they will keep having single connection to each host.

I introduced the opt-in switch so that the user will enable it only when they are sure that the shard-aware port is reachable (described it in the README).

In order to have both zero-conf and robustness in the case described above we could introduce a fallback (similar to what happens for shard-aware connections to bad shard). We could also make it opt-out and/or be OK with the scenario described above, but describe it in some kind of troubleshooting section in README.

WDUT?

Reply:
I think it should use shard aware port if we detect that it is supported by a node otherwise fallback to what we have now.

Otherwise I think it would be better to add opt-out in `ClusterConfig` there is already a bunch of `DisableX` config options we can add anohter.

Reply:
Added DisableShardAwarePort flag.